sdk/appconfiguration/app-configuration/src/internal/tracingHelpers.ts

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import {
+  getTracer,
+  Span,
+  SpanOptions,
+  SpanKind,
+  CanonicalCode,
+} from "@azure/core-tracing";
+
+import {
+  RequestOptionsBase,
+  RestError
+} from "@azure/core-http";
+
+/**
+ * @internal
+ * @ignore
+ */
+export class Spanner<TClient> {
+  tracer = getTracer();
```

Comment:
I think we actually don't want to cache the tracer. The reason being that a user can set the tracer at any time, including after this call. `getTracer` will also always return a tracer (the one the user specified or a NoOp tracer) so we can't check if it simply exists and then cache either.

Reply:
Fixing this - should this same comment be on the method itself?

Reply:
I now get it each time I call createSpan() (what used to be called 'setupSpan'. No caching.